codelists/tests/test_models.py

```diff
@@ -105,6 +106,47 @@ def test_handle_name_cannot_be_reused_by_user(user_codelist):
         )
 
 
+def test_handle_name_only_alphanumeric_hyphen_underscore(codelist):
+    with pytest.raises(
+        ValidationError,
+        match="Codelist names must contain at least one letter or number.",
+    ):
+        handle = Handle.objects.create(
+            codelist=codelist,
+            name="!",
+            slug="new-slug",
+            is_current=True,
+            organisation=codelist.organisation,
+        )
+        handle._meta.get_field("name").clean(handle.name, handle)
+
+
+def test_handle_name_non_blank(codelist):
+    with pytest.raises(
+        ValidationError,
+        match="This field cannot be blank.",
+    ):
+        handle = Handle.objects.create(
+            codelist=codelist,
+            name="",
+            slug="new-slug",
+            is_current=True,
+            organisation=codelist.organisation,
+        )
+        handle._meta.get_field("name").clean(handle.name, handle)
+
+
+def test_handle_name_validation_valid(user_codelist):
+    handle = Handle.objects.create(
+        codelist=user_codelist,
+        name="new-codelist_validname123!,\"'.",
+        slug="new-slug",
+        is_current=True,
+        user=user_codelist.user,
+    )
+    handle._meta.get_field("name").clean(handle.name, handle)
+
```

Comment:
question: Since this is valid, could we also `assert` the expected `name` here?

Reply:
that seems very reasonable